src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11Key.java

```diff
@@ -358,6 +358,24 @@ static SecretKey secretKey(Session session, long keyID, String algorithm,
                     new CK_ATTRIBUTE(CKA_SENSITIVE),
                     new CK_ATTRIBUTE(CKA_EXTRACTABLE),
         });
+
+        if ((SunPKCS11.mysunpkcs11 != null) && (!SunPKCS11.isExportWrapKey)
+           && ("AES".equals(algorithm) || "TripleDES".equals(algorithm))
+        ) {
+            if (attrs[0].getBoolean() || attrs[1].getBoolean() || (attrs[2].getBoolean() == false)) {
+                try {
+                    byte[] key = SunPKCS11.mysunpkcs11.exportKey(session.id(), attrs, keyID);
+                    SecretKey aesSecretKey = new SecretKeySpec(key, algorithm);
```

Comment:
The name `aesSecretKey` is a little too specific (it may be a triple-DES key).

Reply:
Updated.